features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/DashDirectUserAuthFragment.kt

```diff
@@ -0,0 +1,220 @@
+import org.slf4j.LoggerFactory
+
+@FlowPreview
+@ExperimentalCoroutinesApi
+@AndroidEntryPoint
+class DashDirectUserAuthFragment : Fragment(R.layout.fragment_dash_direct_user_auth) {
+    companion object {
+        private val log = LoggerFactory.getLogger(DashDirectUserAuthFragment::class.java)
+    }
+
+    private var currentDirectUserAuthType: DashDirectUserAuthType?=null
+    private val binding by viewBinding(FragmentDashDirectUserAuthBinding::bind)
+    private val exploreViewModel: ExploreViewModel by navGraphViewModels(R.id.explore_dash) { defaultViewModelProviderFactory }
+
+    enum class DashDirectUserAuthType(
+        @StringRes val screenTitle: Int,
+        @StringRes val screenSubtitle: Int,
+        @StringRes val textInputHint: Int
+    ) {
+        CREATE_ACCOUNT(
+            R.string.create_dash_direct_account,
+            R.string.log_in_to_dashdirect_account_desc,
+            R.string.email
+        ),
+        SIGN_IN(
+            R.string.log_in_to_dashdirect_account,
+            R.string.log_in_to_dashdirect_account_desc,
+            R.string.email
+        ),
+        OTP(
+            R.string.enter_one_time_password,
+            R.string.check_your_email_and_enter_the_one_time_password,
+            R.string.password
+        );
+    }
+
+    @RequiresApi(Build.VERSION_CODES.R)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.titleBar.setNavigationOnClickListener {
+            hideKeyboard()
+
+            findNavController().popBackStack()
+        }
+        binding.continueButton.isEnabled = false
+
+        binding.input.postDelayed({
+               showKeyboard()
+            },
+            100
+        )
+
+        currentDirectUserAuthType=   (arguments?.getSerializable("dashDirectUserAuthType") as? DashDirectUserAuthType)?.also {
+             binding.title.setText(it.screenTitle)
+             binding.descLabel.setText(it.screenSubtitle)
+             binding.inputWrapper.setHint(it.textInputHint)
+        }
+
+        binding.input.doOnTextChanged { text, _, _, _ ->
+
+            binding.inputWrapper.isErrorEnabled = false
+
+            if(currentDirectUserAuthType!=DashDirectUserAuthType.OTP )
+            binding.continueButton.isEnabled =isEmail(text)
+            else
+                binding.continueButton.isEnabled = !text.isNullOrEmpty()
+        }
+        requireActivity().window?.decorView?.let { decor ->
+            ViewCompat.setOnApplyWindowInsetsListener(decor) { _, insets ->
+               val showingKeyboard = insets.isVisible(WindowInsetsCompat.Type.ime())
+
+                if (showingKeyboard) {
+                    val systemBarsIMEInsets = insets.getInsets(WindowInsetsCompat.Type.ime())
+                    binding.root.setPadding(0, 0, 0, systemBarsIMEInsets.bottom)
+
+                }
+                insets
+            }
```

Comment:
This doesn't work well right now - the button stays raised if I hide the keyboard.

There is a better way to handle this - set `android:windowSoftInputMode="adjustResize"` for the `WalletActivity`. Unless there are side effects of changing that field, we should use it instead. Did you try it?